### PR TITLE
Ref: update to BWC v3.8.0

### PR DESCRIPTION
**Pull request by Gamboster**
Needs `npm i`
- Update to BWC v3.8.0
- Unnecessary translations for BCH addresses removed
- Unit tests fixed
- Refactor for some functions

### CODE REVIEW

File: src/pages/send/confirm/confirm.ts [hunk 1]
```diff
@@ -208,7 +208,7 @@ export class ConfirmPage extends WalletTabsChild {
       // Use legacy address
```
Comment (matiu) [inline src/pages/send/confirm/confirm.ts]: remove this comment?

File: src/pages/send/confirm/confirm.ts [hunk 2]
```diff
@@ -655,7 +655,7 @@ export class ConfirmPage extends WalletTabsChild {
             // Use legacy address
```
Comment (matiu) [inline src/pages/send/confirm/confirm.ts]: rm comment?

File: src/pages/send/confirm/confirm.ts (hunk 1 above)
Comment (Gamboster) [reply src/pages/send/confirm/confirm.ts]: Done

File: src/pages/send/confirm/confirm.ts (hunk 2 above)
Comment (Gamboster) [reply src/pages/send/confirm/confirm.ts]: Done

**Comment by Gamboster** [comment]
> at import-wallet.ts
> 
> derivationPathByDefault
> 
> need to be m/44'/145'/0' for BCH.
> 
> in case the user change it to 0' and coin = BCH.
> import need to pass
> 'use0forBCH' to `importMnemonic`.
> 
> conversely, if the user set path to 44'/145'/0' coin need
> to be changed to BCH.

@matiu Done. You can review it again, Thanks!

File: src/providers/wallet/wallet.ts [hunk 3]
```diff
@@ -454,8 +455,15 @@ export class WalletProvider {
...
       this.persistenceProvider
         .getLastAddress(wallet.id)
-        .then(addr => {
-          if (!forceNew && addr) return resolve(addr);
+        .then((addr: string) => {
+          if (addr) {
+            // prevent to show legacy address
+            const isBchLegacy =
+              wallet.coin == 'bch' &&
+              ['C', 'H', 'm', 'n'].indexOf(addr.charAt(0)) > -1;
```
Comment (matiu) [inline src/providers/wallet/wallet.ts]: this seems a little difficult to read.

maybe:
 const isBchLegacy = 
             wallet.coin == 'bch' &&  addr.match(/^[CHmn]/;

Comment (matiu) [reply src/providers/wallet/wallet.ts]: not sure...

Comment (Gamboster) [reply src/providers/wallet/wallet.ts]: I agree. Done!

**Pull request merged**